files/routes/posts.py

```diff
@@ -36,6 +35,24 @@
 
 titleheaders = {"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.72 Safari/537.36"}
 
+MAX_TITLE_LENGTH = 500
+MAX_URL_LENGTH = 2048
+MAX_BODY_LENGTH = 20000
+
+# Get request value `val` and ensure it is within length constraints
+# : Returns an either tuple (good_value, error)
+# : TODO it may make sense to do more sanitisation here
+def guarded_value(val, min_len, max_len):
```

Comment:
I definitely like extracting the length and format checks from being all over the place in the current responders to being in just one place here. The only thing that could be better is to use it in even more places, but this is plenty for one PR.

The one thing I don't like as much about this method is that it seems to know too much about the error response. It's a little handy that you can directly return what it returns in the edit_post method. But submit_post has it's own error helper method (yes that's pretty weird), so you have to deconstruct the error message out of this method's response to pass it to that, only for that helper to reconstruct it again. I think it would be better if it just returned the string in the error position - edit_post will have to retain the slightly more complex manually constructing the error response, but submit_post will be able to use it directly with its own error helper.

Reply:
I get what you mean about the error deconstruction. I mostly did that to retain the behavior of the old code. However, it strike me that the custom error method in submit_post exists only to provide nice errors to the user, while the guarded_value checks are all re-checking things that are already guarded by the front-end. The implication of this is that an honest user can't hit these errors, so I don't think we're obliged to provide a convenient error response?

With this in mind, perhaps the best solution (and admittedly what I wanted to do first) would be to return the errors from guarded_value directly in submit_post, as is done in edit_post.

What do you think?

Reply:
P.S. to the above - why am I keen on guarded_value returning errors and not just messages? Because intuitively to me the role of the function (as a guard) is to check for errors in the request, and an error in the context of a web handler is (strictly) the pair of a status code and an error message. If guarded_value is returning raw messages and not errors it seems to me to not be doing the job it claims to do.

Reply:
I'm kinda divided on this; it feels like the kind of thing that's really clean to start with, but turns out to be doing too much work and has to be simplified. For example, I can imagine scenarios where we have to prepend another bit of error text to the beginning.

But maybe that's OK, either long-term or for now.

I think I'm calling this Good Enough and putting it in, but note that I won't be entirely surprised if we have to change it. That's how we learn, though!